### PR TITLE
detail level in unk token warnings during ModularTokenizer encode/encode_list

### PR DESCRIPTION
**Pull request by floccinauc**
Problem: ModularTokenizer encode and encode_list functions write lots of data to output when they encounter UNK tokens.
Solution: verbose parameter that reduces output volume.

### CODE REVIEW

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py [hunk 1]
```diff
@@ -991,6 +996,8 @@ def encode_list(
             pad_type_id (Optional[int], optional): _description_. Defaults to 0. (TODO: raise exception)
             return_overflow_info (Optional[bool], optional): _description_. If True return an additional string with overflow information. Defaults to False.
             on_unknown: (Optional[str], optional): What happens if unknown tokens (i.e. ones mapped to <UNK>) are encountered: 'raise' or 'warn'
+            verbose (Optional[int], optional): verbosity level. 0,1: warning notification, 2: warning with partial data, 3: warning
```
Comment (matanninio) [inline fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:999]: I suggest this be spaced out a little, to allow values in-between,

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py [hunk 2]
```diff
@@ -1082,9 +1089,15 @@ def encode_list(
                     f"Encountered unknown tokens in input starting with {typed_input_list[0].input_string}"
                 )
             elif on_unknown == "warn":
-                warn(
-                    f"Encountered unknown tokens in input starting with {typed_input_list[0].input_string}"
-                )
+                if verbose <= 1:
+                    warning_message = "Encountered unknown tokens in input"
+                elif verbose == 2:
+                    warning_message = f"Encountered unknown tokens in input starting with {typed_input_list[0].input_string[:20]}"
```
Comment (matanninio) [inline fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:1095]: Printing the token[s] can be helpful.  

One option that may be useful is to print each unknown token about once.  We do want to know what the unknown tokens are, but we need to not flood the logs.

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py [hunk 3]
```diff
@@ -1082,9 +1089,15 @@ def encode_list(
                     f"Encountered unknown tokens in input starting with {typed_input_list[0].input_string}"
                 )
             elif on_unknown == "warn":
-                warn(
-                    f"Encountered unknown tokens in input starting with {typed_input_list[0].input_string}"
-                )
+                if verbose <= 1:
+                    warning_message = "Encountered unknown tokens in input"
```
Comment (matanninio) [inline fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:1093]: level 0 should probably be "silent".

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py (hunk 2 above)
Comment (sivanravidos) [reply fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:1095]: Also I see the number of unknown tokens is known so maybe print that? X/Y unknown tokens

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py (hunk 3 above)
Comment (floccinauc) [reply fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:1093]: This wat intentional. I don't want to introduce a completely silent option at this point.

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py (hunk 2 above)
Comment (floccinauc) [reply fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py]: > Also I see the number of unknown tokens is known so maybe print that? X/Y unknown tokens

Added this on verbose 2 and 3

File: fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py (hunk 1 above)
Comment (matanninio) [reply fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:999]: I actually do see a need for in-between here:  For our use on gene expression, we know we will have some unknown tokens (it's pretty inherent to the data), so we can't use 2 or 3, as they will flood the log files, but we don't want to ignore it all together as we may have some wrongly unmatched tokens.  I suspect we need something like "report each new un-mapped token", or better yet, give a report on the unmapped tokens found at the end (of each epoch?)

Comment (floccinauc) [reply fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:999]: This brings us back to the problem of finding which tokens map to UNK. In any case, things like new unmapped tokens and epoch statistics do not fall under tokenizer jurisdiction. This is something you should do outside of it.

Comment (matanninio) [reply fusedrug/data/tokenizer/modulartokenizer/modular_tokenizer.py:999]: we can probably manage by pulling the encoded object from the tokenizer via "key_out_tokenized_object" and extracting the problematic tokens from that.  
so - 0 as "no warning", default is 1?

**Review by matanninio — APPROVED** [review]
just change default verbose level and add quite

**Pull request merged**